src/users/user.service.ts

```diff
@@ -121,4 +120,16 @@ export class UserService {
     const foundUser = await this.userModel.find({ email: emailDto.email }).exec();
     return foundUser.length > 0;
   }
+
+  /**
+   * delete user by email
+   * @param emailDto
+   */
+
+  async deleteUserByEmail(emailDto: CheckEmailDto): Promise<{ message?: string }> {
```

Comment:
What is the difference between this deleteUserByEmail service you wrote and removeUserById one we already have? 

Are you sure we are truly going to delete a user but not using soft delete?

Reply:
I am not sure which one is better. I think the data is still in the database if using soft delete. So some users who didn't registered successfully still can login directly.